### PR TITLE
common: stop using ceph/daemon entrypoint

### PR DESCRIPTION
**Pull request by guits**
major refactor in order to make ceph-ansible use the `ceph/ceph` container image instead of `ceph/daemon`

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-subset_update

**Comment by guits** [comment]
jenkins test centos-container-purge

File: roles/ceph-mon/templates/ceph-mon.service.j2 [hunk 1]
```diff
@@ -48,11 +48,13 @@ ExecStart=/usr/bin/{{ container_binary }} run --rm --name ceph-mon-%i \
...
   -e CEPH_PUBLIC_NETWORK={{ public_network | regex_replace(' ', '') }} \
-  -e CEPH_DAEMON=MON \
   -e CONTAINER_IMAGE={{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} \
   -e TCMALLOC_MAX_TOTAL_THREAD_CACHE_BYTES={{ ceph_tcmalloc_max_total_thread_cache }} \
   {{ ceph_mon_docker_extra_env }} \
-  {{ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}
+  --entrypoint=/usr/bin/ceph-mon \
+  {{ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} \
+  -f --default-log-to-file=false --default-log-to-stderr=true \
+  -i {{ monitor_name }} --mon-data /var/lib/ceph/mon/ceph-{{ monitor_name }} --public-addr {{ _current_monitor_address }}
```
Comment (guits) [inline roles/ceph-mon/templates/ceph-mon.service.j2:57]:
```suggestion
  -i {{ monitor_name }} --mon-data /var/lib/ceph/mon/{{ cluster }}-{{ monitor_name }} --public-addr {{ _current_monitor_address }}
```

Comment (guits) [reply roles/ceph-mon/templates/ceph-mon.service.j2:57]: will be addressed in a separate pr

Comment (guits) [inline roles/ceph-mon/templates/ceph-mon.service.j2:57]:
```suggestion
  -i {{ monitor_name }} --mon-data /var/lib/ceph/mon/{{ cluster }}-{{ monitor_name }} --public-addr {{ _current_monitor_address }}
```

File: roles/ceph-nfs/tasks/pre_requisite_container.yml [hunk 2]
```diff
@@ -39,26 +83,25 @@
...
+#         owner: "root"
+#         group: "root"
+#         mode: "0644"
 
-    - name: reload dbus configuration
-      command: "killall -SIGHUP dbus-daemon"
-  when: ceph_nfs_dynamic_exports | bool
+#     - name: reload dbus configuration
+#       command: "killall -SIGHUP dbus-daemon"
+#   when: ceph_nfs_dynamic_exports | bool
```
Comment (guits) [inline roles/ceph-nfs/tasks/pre_requisite_container.yml:107]: need to figure out what to do with this

File: roles/ceph-nfs/templates/ganesha.conf.j2 [hunk 3]
```diff
@@ -1,9 +1,9 @@
 #jinja2: trim_blocks: "true", lstrip_blocks: "true"
 # {{ ansible_managed }}
 
-{% if ceph_nfs_dynamic_exports | bool and not ceph_nfs_rados_backend | bool %}
-%include /etc/ganesha/export.d/INDEX.conf
-{% endif %}
+# {% if ceph_nfs_dynamic_exports | bool and not ceph_nfs_rados_backend | bool %}
+# %include /etc/ganesha/export.d/INDEX.conf
+# {% endif %}
```
Comment (guits) [inline roles/ceph-nfs/templates/ganesha.conf.j2:6]: ditto

File: roles/ceph-nfs/templates/ceph-nfs.service.j2 [hunk 4]
```diff
@@ -22,25 +22,20 @@ ExecStart=/usr/bin/{{ container_binary }} run --rm --net=host \
...
-  -v /var/lib/ceph:/var/lib/ceph:z \
-  -v /etc/ceph:/etc/ceph:z \
-  -v /var/lib/nfs/ganesha:/var/lib/nfs/ganesha:z \
-  -v /etc/ganesha:/etc/ganesha:z \
-  -v /var/run/ceph:/var/run/ceph:z \
-  -v /var/log/ceph:/var/log/ceph:z \
-  -v /var/log/ganesha:/var/log/ganesha:z \
-  {% if ceph_nfs_dynamic_exports | bool %}
-  --privileged \
-  -v /var/run/dbus/system_bus_socket:/var/run/dbus/system_bus_socket \
```
Comment (guits) [inline roles/ceph-nfs/templates/ceph-nfs.service.j2:35]: need to preserve this line

File: roles/ceph-nfs/tasks/pre_requisite_container.yml (hunk 2 above)
Comment (guits) [reply roles/ceph-nfs/tasks/pre_requisite_container.yml:107]: this has to be uncommented out

File: roles/ceph-nfs/templates/ceph-nfs.service.j2 [hunk 5]
```diff
@@ -22,25 +22,20 @@ ExecStart=/usr/bin/{{ container_binary }} run --rm --net=host \
...
+-v /etc/ceph:/etc/ceph:z \
+-v /var/lib/ceph/radosgw/{{ cluster }}-rgw.{{ ansible_facts['hostname'] }}/keyring:/var/lib/ceph/radosgw/{{ cluster }}-rgw.{{ ansible_facts['hostname'] }}/keyring:z \
+-v /var/lib/ceph/radosgw/{{ cluster }}-nfs.{{ ansible_facts['hostname'] }}/keyring:/etc/ceph/keyring:z \
+-v /etc/ganesha:/etc/ganesha:z \
+-v /var/run/ceph:/var/run/ceph:z \
+-v /var/log/ceph:/var/log/ceph:z \
+-v /var/log/ganesha:/var/log/ganesha:z \
+-v /etc/localtime:/etc/localtime:ro \
+{{ ceph_nfs_docker_extra_env }} \
+--entrypoint=/usr/bin/ganesha.nfsd \
```
Comment (gouthampacha) [inline roles/ceph-nfs/templates/ceph-nfs.service.j2:35]: previously, the "start_nfs" script was starting idmapd as well; won't that be required? Also did this solution support nfsv3? if yes, statd and rpcbind are necessary as well?

Comment (guits) [reply roles/ceph-nfs/templates/ceph-nfs.service.j2:35]: hi @gouthampacha sorry for the late answer.
I've no idea about nfsv3 here but this part needs to be reworked indeed.

**Comment by guits** [comment]
jenkins test centos-container-purge

**Review by asm0deuz — APPROVED** [review]
LGTM

**Pull request merged**